### ISSUE TITLE
Image captions exporting as body text

### ISSUE THREAD

**Issue by andre4s** [alleyinteractive/apple-news#307]
I'm getting image captions exporting as body text elements — they have the `body` role in the article JSON.

| ![](http://i.imgur.com/GoW57Gv.png) | ![](http://i.imgur.com/3wOkCQJ.png) |
| ------------------------------------ | --------------------------------------|
|`{"role":"body","text":"A passionate speaker?","format":"markdown","textStyle":"default-body","layout":"body-layout"}` | `{"role":"body","text":"The dynamic duo of the coalition","format":"markdown","textStyle":"default-body","layout":"body-layout"}` |

[Original post](https://bsqr.es/eUT79) & [JSON](https://gist.github.com/andre4s/62c037b9c37a6c4b29b811f3e0b02d24). Apple News channel isn't public yet.

One potentially significant fact is that I'm using the [Better Image Credits](https://wordpress.org/plugins/better-image-credits/) plugin to overlay attribution/license info — could it be causing issues with the transformation into JSON? The structure for images looks like this:
```
<div id="attachment" class="wp-attachment">
	<div class="credits-container wp-image">
		<img class="wp-image" />
		<div class="credits-overlay" />
	</div>
	<p class="wp-caption-text" />
</div>
```

**Comment by andre4s** [comment]
Have tried deactivating image credits plugin and updating post but JSON still contains `"role":"body"`, `"textStyle":"default-body"` and `"layout":"body-layout"`

**Comment by kevinfodness** [comment]
@andre4s—

It appears that the plugin in question is changing the HTML that is being written, which causes normal parsing of the post by Publish to Apple News to not see your caption as a caption. For captioned images, Publish to Apple News expects the WordPress standard `<figure><figcaption>` tags.

The plugin expects the HTML output to look like this:

```html
<figure...>
    <img.../>
    <figcaption...>Caption text here</figcaption>
</figure>
```

I would recommend disabling the plugin and trying to create a new post with an embedded image with a caption and verifying the output. It's possible that the plugin's changes persist on saved posts even after disabling the plugin.

**Comment by andre4s** [comment]
I've diagnosed the issue: the theme wasn't [registering support for the `figure` and `figcaption` HTML5 markup](https://bsqr.es/VzsXj) so it wasn't Better Image Credits breaking things. I've now registered support manually and everything works — including with the credits overlay. [Tag structure comparison](https://gist.github.com/andre4s/4dac5c7d08c6e60bfcc7b23dcb9818c1).

It might be worth noting this requirement somewhere for people to see. From my research, the `<figure>` structure seems only to appear in the default theme from Twenty Sixteen onwards so I imagine a lot of people will be using themes that'll produce the same error
